kickoff/static/suggestions.js

```diff
@@ -386,7 +428,12 @@ function setupVersionSuggestions(versionElement, versions, buildNumberElement, b
         populateRevisionWithLatest(productName, branchName);
         populatePartial(productName, version, previousBuilds, partialElement);
         populatePartialInfo(productName, version);
-        populateL10nChangesets(productName, version, buildNumber);
+
+        $('#' + productName + '-mozillaRevision').change(function(event) {
+            var revision = $(this).val();
+            populateL10nChangesets(productName, version, buildNumber, branchName, revision);
```

Comment:
I ran ship-it in docker to see if was anything else, and noticed (over there, in the corner cases section) that each time the version is changed we make an extra request. rather than making one eg 
* fresh load of submit_release.html
* add a firefox 57.0.1 and there's a request for http://localhost:5000/releases/devedition-57.0-build1/l10n
* make it 57.0.2 and it's that and http://localhost:5000/releases/devedition-57.0.1-build1/l10n
* add 58.0 and append https://l10n.mozilla.org/shipping/l10n-changesets
* add 59.0 and append https://hg.mozilla.org/releases/mozilla-release/raw-file/0a0a804a5b6f252f12d9808b54ed2a7f6ada27e3/browser/locales/l10n-changesets.json

They appear to run sequentially, but is there a quick way to avoid a maybe-race condition here ? If not I'm happy to get ready for 59.0b1 and loop back later, because changing the version is (hopefully) uncommon.

Reply:
Yeah, that's a long standing issue. Each time the version field is changed, the fields below are too. This includes the mozillaRevision field, which now sends an event to the l10n field. In other words, the issue is more visible than it used to be.

Like you said, I don't think this is blocking 59.0b1. I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1429720 to track the problem.